File: src/brs/web/api/http/handler/GetATDetails.java
package brs.web.api.http.handler;

import brs.SignumException;
import brs.services.ParameterService;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.JSONData;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.Parameters.AT_PARAMETER;
import static brs.web.api.http.common.Parameters.HEIGHT_PARAMETER;

public class GetATDetails extends ApiServlet.JsonRequestHandler {

  private final ParameterService parameterService;

  public GetATDetails(ParameterService parameterService) {
    super(new LegacyDocTag[] {LegacyDocTag.AT}, AT_PARAMETER, HEIGHT_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws SignumException {
    return JSONData.at(parameterService.getAT(req));
  }
}

File: src/main/kotlin/brs/api/http/GetATDetails.kt
package brs.api.http

import brs.api.http.common.JSONData
import brs.api.http.common.Parameters.AT_PARAMETER
import brs.services.AccountService
import brs.services.ParameterService
import com.google.gson.JsonElement
import javax.servlet.http.HttpServletRequest

/**
 * TODO
 */
internal class GetATDetails(
    private val parameterService: ParameterService,
    private val accountService: AccountService
) : APIServlet.JsonRequestHandler(arrayOf(APITag.AT), AT_PARAMETER) {
    override fun processRequest(request: HttpServletRequest): JsonElement {
        return JSONData.at(parameterService.getAT(request), accountService)
    }
}
